libbpf-rs/src/map.rs

```diff
@@ -153,18 +153,41 @@ impl Map {
         let path_c = util::path_to_cstring(path)?;
         let path_ptr = path_c.as_ptr();
 
-        let ret = unsafe { libbpf_sys::bpf_map__pin(self.ptr, path_ptr) };
+        let ret = if self.ptr.is_null() {
+            if self.fd <= 0 {
+                return Err(Error::InvalidInput(format!(
+                    "fd must be greater than 0 and got {}",
+                    self.fd
+                )));
+            }
```

Comment:
Why do you need this `self.fd <= 0` check ?

Reply:
I think you're right, it's not necessary to check `fd`. I will remove the check.